processor/processor.go

```diff
@@ -565,7 +565,9 @@ func enhanceWithTimeFields(event *transformer.TransformerEventT, singularEventMa
 	event.Message["receivedAt"] = receivedAt.Format(misc.RFC3339Milli)
 	event.Message["originalTimestamp"] = originalTimestamp.Format(misc.RFC3339Milli)
 	event.Message["sentAt"] = sentAt.Format(misc.RFC3339Milli)
-	event.Message["timestamp"] = misc.GetChronologicalTimeStamp(receivedAt, sentAt, originalTimestamp).Format(misc.RFC3339Milli)
+	if _, ok := event.Message["timestamp"]; !ok {
```

Comment:
Should we check the data type / format of the string sent by user in timestamp field. Can you verify the segment behavior once like what happens if user sends int, bool, invalid timestamp format, different timestamp formats etc.
(also check if it's not empty string, whitespaces etc.)